src/libmongoc/tests/TestSuite.c

```diff
@@ -176,11 +177,14 @@ TestSuite_Init (TestSuite *suite, const char *name, int argc, char **argv)
       } else if ((0 == strcmp ("-s", argv[i])) ||
                  (0 == strcmp ("--silent", argv[i]))) {
          suite->silent = true;
-      } else if (0 == strcmp ("-l", argv[i])) {
+      } else if ((0 == strcmp ("-l", argv[i])) ||
+                 (0 == strcmp ("--match", argv[i]))) {
+         char *val;
          if (argc - 1 == i) {
-            test_error ("-l requires an argument.");
+            test_error ("%s requires an argument.", argv[i]);
          }
-         suite->testname = bson_strdup (argv[++i]);
+         val = bson_strdup (argv[++i]);
```

Comment:
Is there a corresponding bson_free()?

note: bson_strdup() can return NULL if passed a NULL parameter, but I don't believe that's possible here.